Source/Microsoft.Teams.Apps.FAQPlusPlus/Bots/FaqPlusPlusBot.cs

```diff
@@ -768,30 +760,31 @@ private static Task<TaskModuleResponse> GetTaskModuleResponseAsync(Attachment qu
                 return;
             }
 
-            string text = message.Text?.ToLower()?.Trim() ?? string.Empty;
+            string text = (message.Text ?? string.Empty).Trim();
 
-            switch (text)
+            if (text.Equals(Strings.BotCommandAskExpert, StringComparison.InvariantCultureIgnoreCase) ||
+                text.Equals(Constants.AskAnExpert, StringComparison.InvariantCultureIgnoreCase))
             {
-                case Constants.AskAnExpert:
-                    this.logger.LogInformation("Sending user ask an expert card");
-                    await turnContext.SendActivityAsync(MessageFactory.Attachment(AskAnExpertCard.GetCard())).ConfigureAwait(false);
-                    break;
-
-                case Constants.ShareFeedback:
-                    this.logger.LogInformation("Sending user feedback card");
-                    await turnContext.SendActivityAsync(MessageFactory.Attachment(ShareFeedbackCard.GetCard())).ConfigureAwait(false);
-                    break;
-
-                case Constants.TakeATour:
-                    this.logger.LogInformation("Sending user tour card");
-                    var userTourCards = TourCarousel.GetUserTourCards(this.appBaseUri);
-                    await turnContext.SendActivityAsync(MessageFactory.Carousel(userTourCards)).ConfigureAwait(false);
-                    break;
-
-                default:
-                    this.logger.LogInformation("Sending input to QnAMaker");
-                    await this.GetQuestionAnswerReplyAsync(turnContext, message).ConfigureAwait(false);
-                    break;
+                this.logger.LogInformation("Sending user ask an expert card");
+                await turnContext.SendActivityAsync(MessageFactory.Attachment(AskAnExpertCard.GetCard())).ConfigureAwait(false);
+            }
+            else if (text.Equals(Strings.BotCommandFeedback, StringComparison.InvariantCultureIgnoreCase) ||
+                text.Equals(Constants.ShareFeedback, StringComparison.InvariantCultureIgnoreCase))
+            {
+                this.logger.LogInformation("Sending user feedback card");
+                await turnContext.SendActivityAsync(MessageFactory.Attachment(ShareFeedbackCard.GetCard())).ConfigureAwait(false);
+            }
+            else if (text.Equals(Strings.TakeATourButtonText, StringComparison.InvariantCultureIgnoreCase) ||
```

Comment:
Why TakeATourButtonText, and not BotCommandTour (see above where you use BotCommandFeedback, BotCommandAskAnExpert)

Reply:
Thanks. Updated.